File: src/com/juhnowski/onthology/TheObject.java
package com.juhnowski.onthology;

import java.util.ArrayList;

public class TheObject extends Thing implements Colourless,Unalterable,Subsistent{
    ArrayList<Property> internal;
    public ArrayList<Property> external;
    public TheObjectForm form;
    public LogicForm logicForm;
    public TheObject(ArrayList<Property> internal){
        this.internal = internal;
    }

    public boolean equals(TheObject obj) {
        return logicForm.equals(obj.logicForm);
    }
}

File: src/com/juhnowski/onthology/TheObject.kt
package com.juhnowski.onthology

import com.juhnowski.onthology.logical.LogicForm

import java.util.ArrayList

open class TheObject(internal var internal: ArrayList<Property<*>>) : Thing(), Colourless, Unalterable, Subsistent {
    var external: ArrayList<Property<*>>? = null
    var form: TheObjectForm<*>? = null
    var logicForm: LogicForm<*>? = null

    var sign: Sign<*>

    var realName: ArrayList<Symbol>? = null

    var formalConcept: FormalConcepts? = null

    fun equals(obj: TheObject): Boolean {
        return logicForm == obj.logicForm
    }

    fun signify(sign: Sign<*>) {
        this.sign = sign
    }
}
